plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreModule.java

```diff
@@ -63,9 +63,29 @@ protected void setup(Binder binder)
 
         binder.bind(Key.get(boolean.class, AllowHiveTableRename.class)).toInstance(true);
 
+        validateThriftHttpConfiguration();
         install(new ThriftMetastoreAuthenticationModule());
     }
 
+    private void validateThriftHttpConfiguration()
+    {
+        StaticMetastoreConfig staticMetastoreConfig = buildConfigObject(StaticMetastoreConfig.class);
+        ThriftMetastoreConfig thriftMetastoreConfig = buildConfigObject(ThriftMetastoreConfig.class);
+
+        if (!staticMetastoreConfig.getMetastoreUris().isEmpty()) {
+            boolean isHttpMetastore = staticMetastoreConfig.getMetastoreUris().stream()
+                    .anyMatch(uri -> "http".equalsIgnoreCase(uri.getScheme()) || "https".equalsIgnoreCase(uri.getScheme()));
+
+            if (isHttpMetastore && thriftMetastoreConfig.isImpersonationEnabled()) {
+                throw new IllegalStateException("'hive.metastore.thrift.impersonation.enabled' cannot be set to true while using http/https metastore URIs in 'hive.metastore.uri'");
```

Comment:
`hive.metastore.thrift.impersonation.enabled` is just one of the invalid config properties HTTP interface doesn't support. The initialization should fail for all invalid properties. This will probably require a separate config class.